python/ray/runtime_env.py

```diff
@@ -26,3 +44,455 @@ def get_current_runtime_env():
+
+    All options in the resulting dictionary will have non-None values.
+
+    Currently supported options:
+        py_modules (List[URI]): List of URIs (either in the GCS or external
+            storage), each of which is a zip file that will be unpacked and
+            inserted into the PYTHONPATH of the workers.
+        working_dir (URI): URI (either in the GCS or external storage) of a zip
+            file that will be unpacked in the directory of each task/actor.
+        pip (List[str] | str): Either a list of pip packages, or a string
+            containing the path to a pip requirements.txt file.
+        conda (dict | str): Either the conda YAML config, the name of a
+            local conda env (e.g., "pytorch_p36"), or the path to a conda
+            environment.yaml file.
+            The Ray dependency will be automatically injected into the conda
+            env to ensure compatibility with the cluster Ray. The conda name
+            may be mangled automatically to avoid conflicts between runtime
+            envs.
+            This field cannot be specified at the same time as the 'pip' field.
+            To use pip with conda, please specify your pip dependencies within
+            the conda YAML config:
+            https://conda.io/projects/conda/en/latest/user-guide/tasks/manage-e
+            nvironments.html#create-env-file-manually
+            Examples:
+                {"channels": ["defaults"], "dependencies": ["codecov"]}
+                "pytorch_p36"   # Found on DLAMIs
+        container (dict): Require a given (Docker) container image,
+            The Ray worker process will run in a container with this image.
+            The `worker_path` is the default_worker.py path.
+            The `run_options` list spec is here:
+            https://docs.docker.com/engine/reference/run/
+            Examples:
+                {"image": "anyscale/ray-ml:nightly-py38-cpu",
+                 "worker_path": "/root/python/ray/workers/default_worker.py",
+                 "run_options": ["--cap-drop SYS_ADMIN","--log-level=debug"]}
+        env_vars (dict): Environment variables to set.
+            Examples:
+                {"OMP_NUM_THREADS": "32", "TF_WARNINGS": "none"}
+    """
+
+    known_fields: Set[str] = {
+        "py_modules",
+        "working_dir",
+        "conda",
+        "pip",
+        "container",
+        "excludes",
+        "env_vars",
+        "_ray_release",
+        "_ray_commit",
+        "_inject_current_ray",
+        "plugins",
+        "eager_install",
+    }
+
+    def __init__(
+        self,
+        *,
+        py_modules: Optional[List[str]] = None,
+        working_dir: Optional[str] = None,
+        pip: Optional[List[str]] = None,
+        conda: Optional[Union[Dict[str, str], str]] = None,
+        container: Optional[Dict[str, str]] = None,
+        env_vars: Optional[Dict[str, str]] = None,
+        _validate: bool = True,
+        **kwgs,
```

Comment:
Why do we need`**kwgs` param? It is not strong-typed.

Reply:
Consistent with `ParsedRuntimeEnv`, accept any kwargs.
In later PR, class `RuntimeEnv` will inherit `NamedTuple`, only accept `RuntimeEnv.known_fields`

Reply:
use standard naming please `kwargs`

Reply:
done